### ISSUE TITLE
Unable to export configuration - MicrosoftTeams needs to be updated

### ISSUE THREAD

**Issue by spiderchicke2** [microsoft/Microsoft365DSC#3004]
Hello everyone,

I'm having problems exporting configuration. Latest MicrosoftTeams module installed however.

The following dependencies need updating:
    * MicrosoftTeams
Please run Update-M365DSCDependencies as Administrator.Please run Uninstall-M365DSCOutdatedDependencies.
Partial Export file was saved at: C:\Users\xxxxxxx\AppData\Local\Temp\5\30d7ca71-e750-424d-9627-ce9d75903b94.partial.ps1
The following dependencies need updating:
    * MicrosoftTeams
Please run Update-M365DSCDependencies as Administrator.Please run Uninstall-M365DSCOutdatedDependencies.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.308.1\modules\M365DSCUtil.psm1:1276 char:13
+             throw $ErrorMessage
+             ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (The following d...edDependencies.:String) [], RuntimeException
    + FullyQualifiedErrorId : The following dependencies need updating:
    * MicrosoftTeams
Please run Update-M365DSCDependencies as Administrator.Please run Uninstall-M365DSCOutdatedDependencies.
 

PS C:\Scripts\M365DSC> Find-Module microsoftteams

Version    Name                                Repository           Description                                            
-------    ----                                ----------           -----------                                            
5.0.0      MicrosoftTeams                      PSGallery            Microsoft Teams cmdlets module for Windows PowerShel...

**Comment by andikrueger** [comment]
Could you please share the output of `Get-Module MicrosoftTeams - ListAvailable` ?

**Comment by spiderchicke2** [comment]
hmm

nothing there, empty

**Comment by andikrueger** [comment]
Did you execute the two mentioned cmdlets?

**Comment by spiderchicke2** [comment]
trying to do export

**Comment by spiderchicke2** [comment]
Now I have a different error. Not sure if I should open new issue or can continue here

Connecting to {MicrosoftGraph}...✅
[1/17] Extracting [AADApplication] using {ApplicationSecret}...
    |---[1/2230] XXXXXXXXXXXXXXXXXXXXXGet-MgOauth2PermissionGrant : Could not load file or ass
embly 'Microsoft.Graph.Authentication, 
Version=0.9.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system 
cannot find the file specified.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.308.1\DSCResources\MSFT_AADApplication\MSF
T_AADApplication.psm1:897 char:88
+ ... th2PermissionGrant -Filter "ClientId eq '$($appServicePrincipal.Id)'"
+                                                ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-MgOauth2PermissionGrant_List1], FileNotFoundException
    + FullyQualifiedErrorId : Microsoft.Graph.PowerShell.Cmdlets.GetMgOauth2PermissionGrant_List1

**Comment by spiderchicke2** [comment]
list of app permissions assigned:

<html>
<body>


Microsoft Graph (11) |   |   |   |   |  
-- | -- | -- | -- | -- | --
AdministrativeUnit.Read.All | Application | Read all administrative units | Yes | Granted for  |  
Application.Read.All | Application | Read all applications | Yes | Granted for  |  
Directory.Read.All | Application | Read directory data | Yes | Granted for  |  
EntitlementManagement.Read.All | Application | Read all entitlement management resources | Yes | Granted for  |  
Group.Read.All | Application | Read all groups | Yes | Granted for |  
Organization.Read.All | Application | Read organization information | Yes | Granted for  |  
Policy.Read.All | Application | Read your organization's policies | Yes | Granted for |  
RoleManagement.Read.Directory | Application | Read all directory RBAC settings | Yes | Granted for  |  
RoleManagementPolicy.Read.Directory | Application | Read all policies for privileged role assignments of your company's directory | Yes | Granted for  |  
User.Read | Delegated | Sign in and read user profile | No | Granted for  |  
User.Read.All | Application | Read all users' full profiles | Yes


</body>
</html>

**Comment by andikrueger** [comment]
Please run as local Administrator:

Update-M365DSCModule
Update-M365DSCDependencies 
Uninstall-M365DSCOutdatedDependencies.

**Comment by spiderchicke2** [comment]
PS C:\Users\xxx> Update-M365DSCModule
Checking Microsoft.Graph.Authentication
PS C:\Users\xxx> Update-M365DSCDependencies
PS C:\Users\xxx> Uninstall-M365DSCOutdatedDependencies
Checking Microsoft.Graph.Authentication
PS C:\Users\xxx>


result of script execution (part of it)

    |---[6/2230] APPLICATION_1✅
    |---[7/2230] APPLICATION_2✅
    |---[8/2230] APPLICATION_3Get-MgOauth2PermissionGrant : Cou
ld not load file or assembly 'Microsoft.Graph.Authentication, 
Version=0.9.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system 
cannot find the file specified.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.308.1\DSCResources\MSFT_AADApplication\MSF
T_AADApplication.psm1:897 char:88
+ ... th2PermissionGrant -Filter "ClientId eq '$($appServicePrincipal.Id)'"
+                                                ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-MgOauth2PermissionGrant_List1], FileNotFoundException
    + FullyQualifiedErrorId : Microsoft.Graph.PowerShell.Cmdlets.GetMgOauth2PermissionGrant_List1

**Comment by andikrueger** [comment]
What PS version are you using: $PSVersionTable

Could you please share the output of `Get-Module -ListAvailable` and `Get-InstalledModule`

**Comment by spiderchicke2** [comment]
PS C:\Users\xxx> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.14393.5582
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.5582
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

PS C:\Users\xxx> Get-Module -ListAvailable


    Directory: C:\Users\xxx\Documents\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Manifest   1.1.183.57 MSOnline                            {Get-MsolDevice, Remove-MsolDevice, Enable-MsolDevice, Disable
Script     1.4.7      PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSo


    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Binary     2.0.2.128  AzureAD                             {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-Azur
Binary     2.0.2.106  AzureAD                             {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-Azur
Binary     2.0.2.76   AzureAD                             {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-Azur
Script     1.0.957478 AzureADExporter                     {Connect-AzureADExporter, Export-AzureAD}
Binary     2.0.2.129  AzureADPreview                      {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-Azur
Binary     2.0.2.102  AzureADPreview                      {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-Azur
Binary     2.0.2.89   AzureADPreview                      {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-Azur
Binary     2.0.2.17   AzureADPreview                      {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-Azur
Binary     20.12.4... BinaryTree.Power365                 {Start-BTDiscovery, Get-BTEnvironment, Set-BTEnvironment, Get-
Manifest   1.3.0.6    DSCParser                           ConvertTo-DSCObject
Script     3.1.0      ExchangeOnlineManagement            {Get-ConnectionInformation, Get-EXOCasMailbox, Get-EXOMailbox,
Script     0.11.0     ExchangePowerShell                  {Clear-EPAutoMapping, Clear-EPExchangeLogs, Convert-EPIMCEAEXt
Script     7.1.1      ImportExcel                         {Add-ConditionalFormatting, Add-ExcelChart, Add-ExcelDataValid
Script     6.5.1      ImportExcel                         {Add-ConditionalFormatting, Add-ExcelChart, Add-ExcelDataValid
Manifest   1.21.0     Microsoft.Graph
Manifest   1.2.0      Microsoft.Graph
Manifest   0.9.1      Microsoft.Graph
Script     0.9.1      Microsoft.Graph.Analytics           {Get-MgUserActivityStatistics, Get-MgUserLastSharedMethodInsig
Script     1.22.0     Microsoft.Graph.Applications        {Add-MgApplicationKey, Add-MgApplicationPassword, Add-MgServic
Script     1.23.0     Microsoft.Graph.Authentication      {Connect-MgGraph, Disconnect-MgGraph, Get-MgContext, Get-MgPro
Script     1.22.0     Microsoft.Graph.Authentication      {Connect-MgGraph, Disconnect-MgGraph, Get-MgContext, Get-MgPro
Script     1.21.0     Microsoft.Graph.Bookings            {Get-MgBookingBusiness, Get-MgBookingBusinessAppointment, Get-
Script     1.2.0      Microsoft.Graph.Bookings            {Get-MgBookingBusiness, Get-MgBookingBusinessAppointment, Get-
Script     0.9.1      Microsoft.Graph.Bookings            {Get-MgBookingBusiness, Get-MgBookingBusinessAppointment, Get-
Script     1.21.0     Microsoft.Graph.Calendar            {Get-MgGroupCalendar, Get-MgGroupCalendarMultiValueExtendedPro
Script     1.2.0      Microsoft.Graph.Calendar            {Get-MgGroupCalendar, Get-MgGroupCalendarMultiValueExtendedPro
Script     1.21.0     Microsoft.Graph.ChangeNotifications {Get-MgSubscription, Invoke-MgReauthorizeSubscription, New-MgS
Script     1.2.0      Microsoft.Graph.ChangeNotifications {Get-MgSubscription, New-MgSubscription, Remove-MgSubscription
Script     1.21.0     Microsoft.Graph.CloudCommunications {Add-MgCommunicationCallLargeGalleryView, Clear-MgCommunicatio
Script     1.2.0      Microsoft.Graph.CloudCommunications {Get-MgCommunicationCall, Get-MgCommunicationCallAudioRoutingG
Script     1.21.0     Microsoft.Graph.Compliance          {Add-MgComplianceEdiscoveryCaseCustodianHold, Add-MgCompliance
Script     1.2.0      Microsoft.Graph.Compliance          {Close-MgComplianceEdiscoveryCase, Get-MgCompliance, Get-MgCom
Script     1.21.0     Microsoft.Graph.CrossDeviceExper... {Get-MgUserActivity, Get-MgUserActivityHistoryItem, Get-MgUser
Script     1.2.0      Microsoft.Graph.CrossDeviceExper... {Get-MgUserActivity, Get-MgUserActivityHistoryItem, Get-MgUser
Script     1.22.0     Microsoft.Graph.DeviceManagement    {Get-MgDeviceManagement, Get-MgDeviceManagementAdvancedThreatP
Script     1.21.0     Microsoft.Graph.DeviceManagement... {Add-MgDeviceManagementGroupPolicyUploadedDefinitionFileLangua
Script     1.22.0     Microsoft.Graph.DeviceManagement... {Get-MgDeviceManagementApplePushNotificationCertificate, Get-M
Script     1.22.0     Microsoft.Graph.DeviceManagement... {Get-MgDeviceManagementAndroidDeviceOwnerEnrollmentProfile, Ge
Script     1.21.0     Microsoft.Graph.DeviceManagement... {Compare-MgDeviceManagementIntent, Compare-MgDeviceManagementT
Script     1.21.0     Microsoft.Graph.Devices.CloudPrint  {Get-MgPrint, Get-MgPrintConnector, Get-MgPrintOperation, Get-
Script     1.2.0      Microsoft.Graph.Devices.CloudPrint  {Get-MgPrint, Get-MgPrintConnector, Get-MgPrintOperation, Get-
Script     1.22.0     Microsoft.Graph.Devices.Corporat... {Clear-MgDeviceAppMgtWindowInformationProtectionDeviceRegistra
Script     1.21.0     Microsoft.Graph.Devices.ServiceA... {Get-MgServiceAnnouncementHealthOverview, Get-MgServiceAnnounc
Script     0.9.1      Microsoft.Graph.DevicesApps.Devi... {Clear-MgDeviceAppMgtWindowInformationProtectionDeviceRegistra
Script     0.9.1      Microsoft.Graph.DevicesApps.Mobi... {Get-MgUserManagedAppRegistration, Get-MgUserManagedAppRegistr
Script     0.9.1      Microsoft.Graph.DevicesApps.Offi... {Get-MgOfficeClientConfiguration, Get-MgOfficeClientConfigurat
Script     0.9.1      Microsoft.Graph.DevicesApps.Shar... {Get-MgUserMobileAppTroubleshootingEvent, Get-MgUserMobileAppT
Script     1.21.0     Microsoft.Graph.DirectoryObjects    {Confirm-MgDirectoryObjectMemberGroup, Confirm-MgDirectoryObje
Script     1.2.0      Microsoft.Graph.DirectoryObjects    {Confirm-MgDirectoryObjectMemberGroup, Confirm-MgDirectoryObje
Script     1.21.0     Microsoft.Graph.Education           {Get-MgEducationClass, Get-MgEducationClassAssignment, Get-MgE
Script     1.2.0      Microsoft.Graph.Education           {Get-MgEducationClass, Get-MgEducationClassAssignment, Get-MgE
Script     0.9.1      Microsoft.Graph.Education           {Get-MgEducationClass, Get-MgEducationClassAssignment, Get-MgE
Script     1.21.0     Microsoft.Graph.Files               {Add-MgDriveListContentTypeCopy, Add-MgDriveListContentTypeCop
Script     1.2.0      Microsoft.Graph.Files               {Get-MgDrive, Get-MgDriveActivity, Get-MgDriveBundle, Get-MgDr
Script     0.9.1      Microsoft.Graph.Files.Drives        {Get-MgDrive, Get-MgDriveActivity, Get-MgDriveActivityDriveIte
Script     0.9.1      Microsoft.Graph.Files.Permissions   {Get-MgGroupPermissionGrant, New-MgGroupPermissionGrant, Remov
Script     0.9.1      Microsoft.Graph.Files.Shares        {Get-MgShareDriveItem, Get-MgShareItem, Get-MgShareList, Get-M
Script     1.21.0     Microsoft.Graph.Financials          {Get-MgFinancial, Get-MgFinancialCompany, Get-MgFinancialCompa
Script     1.2.0      Microsoft.Graph.Financials          {Get-MgFinancial, Get-MgFinancialCompany, Get-MgFinancialCompa
Script     0.9.1      Microsoft.Graph.Financials          {Get-MgFinancial, Get-MgFinancialCompany, Get-MgFinancialCompa
Script     1.22.0     Microsoft.Graph.Groups              {Add-MgGroupDriveListContentTypeCopy, Add-MgGroupDriveListCont
Script     0.9.1      Microsoft.Graph.Groups.Actions      {Add-MgGroupFavorite, Confirm-MgGroupGrantedPermission, Confir
Script     0.9.1      Microsoft.Graph.Groups.Calendar     {Get-MgGroupCalendar, Get-MgGroupCalendarEvent, Get-MgGroupCal
Script     0.9.1      Microsoft.Graph.Groups.Conversation {Get-MgGroupConversation, Get-MgGroupConversationThread, Get-M
Script     0.9.1      Microsoft.Graph.Groups.Conversat... {Get-MgGroupThread, Get-MgGroupThreadPost, Get-MgGroupThreadPo
Script     0.9.1      Microsoft.Graph.Groups.Directory... {Get-MgGroupAcceptedSender, Get-MgGroupCreatedOnBehalf, Get-Mg
Script     0.9.1      Microsoft.Graph.Groups.Drive        {Get-MgGroupDrive, New-MgGroupDrive, Remove-MgGroupDrive, Upda
Script     0.9.1      Microsoft.Graph.Groups.Endpoint     {Get-MgGroupEndpoint, New-MgGroupEndpoint, Remove-MgGroupEndpo
Script     0.9.1      Microsoft.Graph.Groups.Extension    {Get-MgGroupExtension, New-MgGroupExtension, Remove-MgGroupExt
Script     0.9.1      Microsoft.Graph.Groups.Functions    {Get-MgGroupCalendarEventDelta, Get-MgGroupCalendarEventInstan
Script     0.9.1      Microsoft.Graph.Groups.Group        {Get-MgGroup, New-MgGroup, Remove-MgGroup, Update-MgGroup}
Script     0.9.1      Microsoft.Graph.Groups.Lifecycle... {Add-MgGroupToLifecyclePolicy, Get-MgGroupLifecyclePolicy, Inv
Script     0.9.1      Microsoft.Graph.Groups.OneNote      {Get-MgGroupOnenoteNotebook, Get-MgGroupOnenoteNotebookSection
Script     0.9.1      Microsoft.Graph.Groups.Planner      {Get-MgGroupPlanner, Get-MgGroupPlannerPlan, Get-MgGroupPlanne
Script     0.9.1      Microsoft.Graph.Groups.ProfilePhoto {Get-MgGroupPhoto, New-MgGroupPhoto, Remove-MgGroupPhoto, Upda
Script     0.9.1      Microsoft.Graph.Groups.Site         {Get-MgGroupSite, New-MgGroupSite, Remove-MgGroupSite, Update-
Script     0.9.1      Microsoft.Graph.Identity.AccessR... {Add-MgAccessReviewDecision, Get-MgAccessReview, Get-MgAccessR
Script     0.9.1      Microsoft.Graph.Identity.Adminis... {Confirm-MgAdministrativeUnitMemberGroup, Confirm-MgAdministra
Script     0.9.1      Microsoft.Graph.Identity.Applica... {Add-MgApplicationKey, Add-MgApplicationPassword, Confirm-MgAp
Script     0.9.1      Microsoft.Graph.Identity.AppRole... {Get-MgGroupAppRoleAssignment, Get-MgUserAppRoleAssignment, Ne
Script     0.9.1      Microsoft.Graph.Identity.AuditLogs  {Get-MgAuditLogDirectoryAudit, Get-MgAuditLogDirectoryProvisio
Script     0.9.1      Microsoft.Graph.Identity.Authent... {Get-MgUserAuthentication, Get-MgUserAuthenticationEmailMethod
Script     0.9.1      Microsoft.Graph.Identity.AzureADPIM {Get-MgPrivilegedApproval, Get-MgPrivilegedApprovalRequest, Ge
Script     0.9.1      Microsoft.Graph.Identity.Conditi... {Get-MgIdentityConditionalAccess, Get-MgIdentityConditionalAcc
Script     0.9.1      Microsoft.Graph.Identity.Contracts  {Confirm-MgContractMemberGroup, Confirm-MgContractMemberObject
Script     0.9.1      Microsoft.Graph.Identity.DataPol... {Get-MgDataPolicyOperation, New-MgDataPolicyOperation, Remove-
Script     0.9.1      Microsoft.Graph.Identity.Devices    {Confirm-MgDeviceMemberGroup, Confirm-MgDeviceMemberObject, Ge
Script     0.9.1      Microsoft.Graph.Identity.Directory  {Get-MgDirectory, Get-MgDirectoryDeletedItem, Get-MgDirectoryF
Script     1.22.0     Microsoft.Graph.Identity.Directo... {Complete-MgDirectoryImpactedResource, Complete-MgDirectoryRec
Script     0.9.1      Microsoft.Graph.Identity.Directo... {Confirm-MgDirectoryObjectMemberGroup, Confirm-MgDirectoryObje
Script     0.9.1      Microsoft.Graph.Identity.Directo... {Confirm-MgDirectoryRoleMemberGroup, Confirm-MgDirectoryRoleMe
Script     0.9.1      Microsoft.Graph.Identity.Directo... {Confirm-MgDirectoryRoleTemplateMemberGroup, Confirm-MgDirecto
Script     0.9.1      Microsoft.Graph.Identity.Directo... {Get-MgDirectorySetting, Get-MgGroupSetting, New-MgDirectorySe
Script     0.9.1      Microsoft.Graph.Identity.Directo... {Confirm-MgDirectorySettingTemplateMemberGroup, Confirm-MgDire
Script     0.9.1      Microsoft.Graph.Identity.Domains    {Get-MgDomain, Get-MgDomainNameerenceByRef, Get-MgDomainRefDom
Script     1.22.0     Microsoft.Graph.Identity.Governance {Add-MgAccessReviewDecision, Add-MgAccessReviewInstanceDecisio
Script     0.9.1      Microsoft.Graph.Identity.Invitat... {Get-MgInvitation, Get-MgInvitationInvitedUser, Get-MgInvitati
Script     0.9.1      Microsoft.Graph.Identity.OAuth2P... {Get-MgOauth2PermissionGrant, Get-MgOauth2PermissionGrantDelta
Script     0.9.1      Microsoft.Graph.Identity.OnPremi... {Get-MgOnPremisePublishingProfile, Get-MgOnPremisePublishingPr
Script     0.9.1      Microsoft.Graph.Identity.Organiz... {Confirm-MgOrganizationMemberGroup, Confirm-MgOrganizationMemb
Script     0.9.1      Microsoft.Graph.Identity.Organiz... {Confirm-MgContactMemberGroup, Confirm-MgContactMemberObject,
Script     0.9.1      Microsoft.Graph.Identity.Policies   {Get-MgPolicyActivityBasedTimeoutPolicy, Get-MgPolicyAdminCons
Script     0.9.1      Microsoft.Graph.Identity.Protection {Confirm-MgRiskyUserCompromised, Get-MgRiskDetection, Get-MgRi
Script     0.9.1      Microsoft.Graph.Identity.Providers  {Get-MgIdentityProvider, Invoke-MgAvailableIdentityProviderTyp
Script     0.9.1      Microsoft.Graph.Identity.RoleMan... {Get-MgRoleManagement, Get-MgRoleManagementDeviceManagement, G
Script     0.9.1      Microsoft.Graph.Identity.Service... {Add-MgServicePrincipalKey, Add-MgServicePrincipalPassword, Co
Script     1.22.0     Microsoft.Graph.Identity.SignIns    {Confirm-MgInformationProtectionSignature, Confirm-MgRiskyServ
Script     0.9.1      Microsoft.Graph.Identity.Subscri... {Get-MgSubscribedSku, New-MgSubscribedSku, Remove-MgSubscribed
Script     0.9.1      Microsoft.Graph.Identity.TermsOfUse {Get-MgAgreement, Get-MgAgreementAcceptance, Get-MgAgreementFi
Script     0.9.1      Microsoft.Graph.Identity.TrustFr... {Get-MgTrustFramework, Get-MgTrustFrameworkKeySet, Get-MgTrust
Script     0.9.1      Microsoft.Graph.Identity.UserFlows  {Get-MgIdentityUserFlow, New-MgIdentityUserFlow, Remove-MgIden
Script     1.21.0     Microsoft.Graph.Mail                {Get-MgUserInferenceClassification, Get-MgUserInferenceClassif
Script     1.2.0      Microsoft.Graph.Mail                {Get-MgUserInferenceClassification, Get-MgUserInferenceClassif
Script     1.21.0     Microsoft.Graph.ManagedTenants      {Get-MgTenantRelationshipManagedTenant, Get-MgTenantRelationsh
Script     1.21.0     Microsoft.Graph.Notes               {Get-MgGroupOnenoteNotebook, Get-MgGroupOnenoteNotebookSection
Script     1.2.0      Microsoft.Graph.Notes               {Get-MgGroupOnenoteNotebook, Get-MgGroupOnenoteNotebookSection
Script     0.9.1      Microsoft.Graph.Notification        {Get-MgUserNotification, New-MgUserNotification, Remove-MgUser
Script     0.9.1      Microsoft.Graph.OnlineMeetings      {Get-MgUserOnlineMeeting, New-MgUserOnlineMeeting, Remove-MgUs
Script     1.21.0     Microsoft.Graph.People              {Get-MgUserActivityStatistics, Get-MgUserLastSharedMethodInsig
Script     1.2.0      Microsoft.Graph.People              {Get-MgUserActivityStatistics, Get-MgUserLastSharedMethodInsig
Script     1.21.0     Microsoft.Graph.PersonalContacts    {Get-MgUserContact, Get-MgUserContactExtension, Get-MgUserCont
Script     1.2.0      Microsoft.Graph.PersonalContacts    {Get-MgUserContact, Get-MgUserContactExtension, Get-MgUserCont
Script     0.9.1      Microsoft.Graph.Places              {Get-MgPlace, New-MgPlace, Remove-MgPlace, Update-MgPlace}
Script     1.22.0     Microsoft.Graph.Planner             {Get-MgGroupPlanner, Get-MgGroupPlannerPlan, Get-MgGroupPlanne
Script     1.21.0     Microsoft.Graph.Reports             {Confirm-MgAuditLogSignInCompromised, Confirm-MgAuditLogSignIn
Script     1.2.0      Microsoft.Graph.Reports             {Get-MgAuditLogDirectoryAudit, Get-MgAuditLogDirectoryProvisio
Script     0.9.1      Microsoft.Graph.Reports             {Get-MgReportApplicationSign, Get-MgReportAzureAdApplicationSi
Script     1.21.0     Microsoft.Graph.SchemaExtensions    {Get-MgSchemaExtension, New-MgSchemaExtension, Remove-MgSchema
Script     1.2.0      Microsoft.Graph.SchemaExtensions    {Get-MgSchemaExtension, New-MgSchemaExtension, Remove-MgSchema
Script     0.9.1      Microsoft.Graph.SchemaExtensions    {Get-MgSchemaExtension, New-MgSchemaExtension, Remove-MgSchema
Script     1.21.0     Microsoft.Graph.Search              {Add-MgExternalConnectionItemActivity, Get-MgExternal, Get-MgE
Script     1.2.0      Microsoft.Graph.Search              {Get-MgExternal, Get-MgExternalConnection, Get-MgSearchEntity,
Script     0.9.1      Microsoft.Graph.Search              {Get-MgExternal, Get-MgExternalConnection, Get-MgSearch, Invok
Script     1.21.0     Microsoft.Graph.Security            {Add-MgSecurityCaseEdiscoveryCaseCustodianHold, Add-MgSecurity
Script     1.2.0      Microsoft.Graph.Security            {Get-MgSecurityAction, Get-MgSecurityAlert, Get-MgSecurityClou
Script     0.9.1      Microsoft.Graph.Security            {Get-MgSecurityAction, Get-MgSecurityAlert, Get-MgSecurityClou
Script     1.21.0     Microsoft.Graph.Sites               {Add-MgSiteContentTypeCopy, Add-MgSiteContentTypeCopyFromConte
Script     1.2.0      Microsoft.Graph.Sites               {Copy-MgSiteOnenoteNotebook, Copy-MgSiteOnenotePageToSection,
Script     0.9.1      Microsoft.Graph.Sites.Actions       {Copy-MgSiteOnenoteNotebook, Copy-MgSiteOnenotePageToSection,
Script     0.9.1      Microsoft.Graph.Sites.Drive         {Get-MgSiteDrive, New-MgSiteDrive, Remove-MgSiteDrive, Update-
Script     0.9.1      Microsoft.Graph.Sites.Functions     {Get-MgSiteActivityByInterval, Get-MgSiteByPath, Get-MgSiteLis
Script     0.9.1      Microsoft.Graph.Sites.List          {Get-MgSiteList, Get-MgSiteListActivity, Get-MgSiteListActivit
Script     0.9.1      Microsoft.Graph.Sites.OneNote       {Get-MgSiteOnenoteNotebook, Get-MgSiteOnenoteNotebookSection,
Script     0.9.1      Microsoft.Graph.Sites.Pages         {Get-MgSitePage, New-MgSitePage, Remove-MgSitePage, Update-MgS
Script     0.9.1      Microsoft.Graph.Sites.Site          {Get-MgSite, Get-MgSiteAnalytic, Get-MgSiteAnalyticByRef, Get-
Script     0.9.1      Microsoft.Graph.Subscriptions       {Get-MgSubscription, New-MgSubscription, Remove-MgSubscription
Script     1.22.0     Microsoft.Graph.Teams               {Add-MgChatMember, Add-MgTeamChannelMember, Add-MgTeamMember,
Script     0.9.1      Microsoft.Graph.Teams.AppCatalogs   {Get-MgAppCatalog, Get-MgAppCatalogTeamApp, Get-MgAppCatalogTe
Script     0.9.1      Microsoft.Graph.Teams.Chats         {Get-MgChat, Get-MgChatInstalledApp, Get-MgChatInstalledAppTea
Script     0.9.1      Microsoft.Graph.Teams.Team          {Copy-MgTeam, Get-MgGroupTeam, Get-MgTeam, Get-MgTeamChannel..
Script     0.9.1      Microsoft.Graph.Teams.Teamwork      {Get-MgUserTeamwork, Get-MgUserTeamworkInstalledApp, Get-MgUse
Script     1.22.0     Microsoft.Graph.Users               {Get-MgUser, Get-MgUserCreatedObject, Get-MgUserDirectReport,
Script     1.22.0     Microsoft.Graph.Users.Actions       {Add-MgUserChatMember, Add-MgUserDriveListContentTypeCopy, Add
Script     0.9.1      Microsoft.Graph.Users.ActivityFeed  {Get-MgUserActivity, Get-MgUserActivityHistoryItem, Get-MgUser
Script     0.9.1      Microsoft.Graph.Users.Calendar      {Get-MgUserCalendar, Get-MgUserCalendarEvent, Get-MgUserCalend
Script     0.9.1      Microsoft.Graph.Users.Contacts      {Get-MgUserContact, Get-MgUserContactExtension, Get-MgUserCont
Script     0.9.1      Microsoft.Graph.Users.Devices       {Get-MgUserDevice, New-MgUserDevice, Remove-MgUserDevice, Upda
Script     0.9.1      Microsoft.Graph.Users.DirectoryO... {Get-MgUserCreatedObject, Get-MgUserCreatedObjectByRef, Get-Mg
Script     0.9.1      Microsoft.Graph.Users.Drive         {Get-MgUserDrive, New-MgUserDrive, Remove-MgUserDrive, Update-
Script     0.9.1      Microsoft.Graph.Users.Extensions    {Get-MgUserExtension, New-MgUserExtension, Remove-MgUserExtens
Script     0.9.1      Microsoft.Graph.Users.FollowedSites {Get-MgUserFollowedSite, Get-MgUserFollowedSiteByRef, New-MgUs
Script     1.21.0     Microsoft.Graph.Users.Functions     {Export-MgUserDeviceAndAppManagementData, Find-MgUserRoom, Fin
Script     1.2.0      Microsoft.Graph.Users.Functions     {Export-MgUserDeviceAndAppManagementData623C, Export-MgUserDev
Script     0.9.1      Microsoft.Graph.Users.Functions     {Export-MgUserDeviceAndAppManagementData623C, Export-MgUserDev
Script     0.9.1      Microsoft.Graph.Users.Groups        {Get-MgUserJoinedGroup, Get-MgUserJoinedTeam, New-MgUserJoined
Script     0.9.1      Microsoft.Graph.Users.Informatio... {Get-MgUserInformationProtection, Remove-MgUserInformationProt
Script     0.9.1      Microsoft.Graph.Users.LicenseDet... {Get-MgUserLicenseDetail, New-MgUserLicenseDetail, Remove-MgUs
Script     0.9.1      Microsoft.Graph.Users.Mail          {Get-MgUserInferenceClassification, Get-MgUserInferenceClassif
Script     0.9.1      Microsoft.Graph.Users.OneNote       {Get-MgUserOnenoteNotebook, Get-MgUserOnenoteNotebookSection,
Script     0.9.1      Microsoft.Graph.Users.OutlookUser   {Get-MgUserOutlook, Get-MgUserOutlookMasterCategory, Get-MgUse
Script     0.9.1      Microsoft.Graph.Users.People        {Get-MgUserPerson, New-MgUserPerson, Remove-MgUserPerson, Upda
Script     0.9.1      Microsoft.Graph.Users.Planner       {Get-MgUserPlanner, Get-MgUserPlannerFavoritePlan, Get-MgUserP
Script     0.9.1      Microsoft.Graph.Users.ProfilePhoto  {Get-MgUserPhoto, New-MgUserPhoto, Remove-MgUserPhoto, Update-
Script     0.9.1      Microsoft.Graph.Users.User          {Get-MgUser, Get-MgUserPresence, New-MgUser, Remove-MgUser...}
Script     0.9.1      Microsoft.Graph.Users.UserSettings  {Get-MgUserSetting, Get-MgUserSettingRegionalAndLanguageSettin
Script     1.21.0     Microsoft.Graph.WindowsUpdates      {Add-MgWindowsUpdatesDeploymentAudienceExclusionMemberById, Ad
Binary     16.0.22... Microsoft.Online.SharePoint.Powe... {Register-SPODataEncryptionPolicy, Update-SPODataEncryptionPol
Binary     16.0.20... Microsoft.Online.SharePoint.Powe... {Register-SPODataEncryptionPolicy, Update-SPODataEncryptionPol
Binary     16.0.20... Microsoft.Online.SharePoint.Powe... {Register-SPODataEncryptionPolicy, Update-SPODataEncryptionPol
Script     2.0.155    Microsoft.PowerApps.Administrati... {New-AdminPowerAppCdsDatabase, Get-AdminPowerAppCdsDatabaseLan
Script     1.0.1      Microsoft.PowerShell.Operation.V... {Get-OperationValidation, Invoke-OperationValidation}
Manifest   1.23.308.1 Microsoft365DSC                     {Assert-M365DSCBlueprint, Compare-M365DSCConfigurations, Confi
Script     5.0.0      MicrosoftTeams                      {Add-TeamChannelUser, Add-TeamUser, Connect-MicrosoftTeams, Di
Script     1.0.105    MSCloudLoginAssistant               {Test-MSCloudLogin, Connect-M365Tenant, Invoke-MSCloudLoginMic
Script     1.3.0.0    MSGraph                             {Get-MgaMailFolder, Rename-MgaMailFolder, New-MgaMailFolder, M
Manifest   1.1.183.57 MSOnline                            {Get-MsolDevice, Remove-MsolDevice, Enable-MsolDevice, Disable
Manifest   1.1.183.17 MSOnline                            {Get-MsolDevice, Remove-MsolDevice, Enable-MsolDevice, Disable
Script     1.0.51     MSOnlineExt                         {Get-MsolTenantContext, Get-MSOnlineExtTelemetryOption, Remove
Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSo
Script     3.4.0      Pester                              {Describe, Context, It, Should...}
Manifest   1.12.0     PnP.PowerShell                      {Add-PnPAdaptiveScopeProperty, Add-PnPPropertyBagValue, Add-Pn
Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}
Script     1.7.249    PSFramework                         {ConvertTo-PSFHashtable, Invoke-PSFCallback, Invoke-PSFProtect
Script     1.2        PSReadline                          {Get-PSReadlineKeyHandler, Set-PSReadlineKeyHandler, Remove-PS
Manifest   2.0.0.14   ReverseDSC                          {Get-DSCParamType, Get-DSCBlock, Get-DSCFakeParameters, Get-DS


    Directory: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Manifest   1.0.0.0    ActiveDirectory                     {Add-ADCentralAccessPolicyMember, Add-ADComputerServiceAccount
Manifest   1.0.0.0    AppBackgroundTask                   {Disable-AppBackgroundTaskDiagnosticLog, Enable-AppBackgroundT
Manifest   2.0.0.0    AppLocker                           {Get-AppLockerFileInformation, Get-AppLockerPolicy, New-AppLoc
Manifest   1.0.0.0    AppvClient                          {Add-AppvClientConnectionGroup, Add-AppvClientPackage, Add-App
Manifest   2.0.0.0    Appx                                {Add-AppxPackage, Get-AppxPackage, Get-AppxPackageManifest, Re
Script     1.0.0.0    AssignedAccess                      {Clear-AssignedAccess, Get-AssignedAccess, Set-AssignedAccess}
Manifest   1.0        BestPractices                       {Get-BpaModel, Get-BpaResult, Invoke-BpaModel, Set-BpaResult}
Manifest   2.0.0.0    BitsTransfer                        {Add-BitsFile, Complete-BitsTransfer, Get-BitsTransfer, Remove
Manifest   1.0.0.0    BranchCache                         {Add-BCDataCacheExtension, Clear-BCCache, Disable-BC, Disable-
Manifest   1.0.0.0    CimCmdlets                          {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, Get
Manifest   1.0        ConfigCI                            {Get-SystemDriver, New-CIPolicyRule, New-CIPolicy, Get-CIPolic
Manifest   1.0        DFSN                                {Get-DfsnRoot, Remove-DfsnRoot, Set-DfsnRoot, New-DfsnRoot...}
Binary     2.0.0.0    DFSR                                {New-DfsReplicationGroup, Get-DfsReplicationGroup, Set-DfsRepl
Manifest   2.0.0.0    DhcpServer                          {Add-DhcpServerInDC, Add-DhcpServerv4Class, Add-DhcpServerv4Ex
Manifest   1.0.0.0    DirectAccessClientComponents        {Disable-DAManualEntryPointSelection, Enable-DAManualEntryPoin
Script     3.0        Dism                                {Add-AppxProvisionedPackage, Add-WindowsDriver, Add-WindowsCap
Manifest   1.0.0.0    DnsClient                           {Resolve-DnsName, Clear-DnsClientCache, Get-DnsClient, Get-Dns
Manifest   2.0.0.0    DnsServer                           {Add-DnsServerConditionalForwarderZone, Add-DnsServerDirectory
Manifest   1.0.0.0    EventTracingManagement              {New-EtwTraceSession, Get-EtwTraceSession, Set-EtwTraceSession
Manifest   1.0.0.0    GroupPolicy                         {Backup-GPO, Block-GPInheritance, Copy-GPO, Get-GPInheritance.
Script     1.0.0.0    IISAdministration                   {Get-IISAppPool, Start-IISCommitDelay, Stop-IISCommitDelay, Ge
Manifest   2.0.0.0    International                       {Get-WinDefaultInputMethodOverride, Set-WinDefaultInputMethodO
Manifest   1.0.0.0    iSCSI                               {Get-IscsiTargetPortal, New-IscsiTargetPortal, Remove-IscsiTar
Manifest   2.0.0.0    IscsiTarget                         {Add-ClusteriSCSITargetServerRole, Add-IscsiVirtualDiskTargetM
Script     1.0.0.0    ISE                                 {New-IseSnippet, Import-IseSnippet, Get-IseSnippet}
Manifest   1.0.0.0    Kds                                 {Add-KdsRootKey, Get-KdsRootKey, Test-KdsRootKey, Set-KdsConfi
Manifest   1.0.1.0    Microsoft.PowerShell.Archive        {Compress-Archive, Expand-Archive}
Manifest   3.0.0.0    Microsoft.PowerShell.Diagnostics    {Get-WinEvent, Get-Counter, Import-Counter, Export-Counter...}
Manifest   3.0.0.0    Microsoft.PowerShell.Host           {Start-Transcript, Stop-Transcript}
Manifest   1.0.0.0    Microsoft.PowerShell.LocalAccounts  {Add-LocalGroupMember, Disable-LocalUser, Enable-LocalUser, Ge
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-ItemProperty, Join-Path...}
Script     1.0        Microsoft.PowerShell.ODataUtils     Export-ODataEndpointProxy
Manifest   3.0.0.0    Microsoft.PowerShell.Security       {Get-Acl, Set-Acl, Get-PfxCertificate, Get-Credential...}
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Format-List, Format-Custom, Format-Table, Format-Wide...}
Manifest   3.0.0.0    Microsoft.WSMan.Management          {Disable-WSManCredSSP, Enable-WSManCredSSP, Get-WSManCredSSP,
Manifest   1.0        MMAgent                             {Disable-MMAgent, Enable-MMAgent, Set-MMAgent, Get-MMAgent...}
Manifest   1.0.0.0    MsDtc                               {New-DtcDiagnosticTransaction, Complete-DtcDiagnosticTransacti
Manifest   2.0.0.0    NetAdapter                          {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-NetAda
Manifest   1.0.0.0    NetConnection                       {Get-NetConnectionProfile, Set-NetConnectionProfile}
Manifest   1.0.0.0    NetEventPacketCapture               {New-NetEventSession, Remove-NetEventSession, Get-NetEventSess
Manifest   2.0.0.0    NetLbfo                             {Add-NetLbfoTeamMember, Add-NetLbfoTeamNic, Get-NetLbfoTeam, G
Manifest   1.0.0.0    NetNat                              {Get-NetNat, Get-NetNatExternalAddress, Get-NetNatStaticMappin
Manifest   2.0.0.0    NetQos                              {Get-NetQosPolicy, Set-NetQosPolicy, Remove-NetQosPolicy, New-
Manifest   2.0.0.0    NetSecurity                         {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPsecMai
Manifest   1.0.0.0    NetSwitchTeam                       {New-NetSwitchTeam, Remove-NetSwitchTeam, Get-NetSwitchTeam, R
Manifest   1.0.0.0    NetTCPIP                            {Get-NetIPAddress, Get-NetIPInterface, Get-NetIPv4Protocol, Ge
Manifest   1.0.0.0    NetworkConnectivityStatus           {Get-DAConnectionStatus, Get-NCSIPolicyConfiguration, Reset-NC
Manifest   1.0.0.0    NetworkSwitchManager                {Disable-NetworkSwitchEthernetPort, Enable-NetworkSwitchEthern
Manifest   1.0.0.0    NetworkTransition                   {Add-NetIPHttpsCertBinding, Disable-NetDnsTransitionConfigurat
Manifest   1.0        NFS                                 {Get-NfsMappedIdentity, Get-NfsNetgroup, Install-NfsMappingSto
Manifest   1.0.0.0    PcsvDevice                          {Get-PcsvDevice, Start-PcsvDevice, Stop-PcsvDevice, Restart-Pc
Manifest   1.0.0.0    PKI                                 {Add-CertificateEnrollmentPolicyServer, Export-Certificate, Ex
Manifest   1.0.0.0    PlatformIdentifier                  Get-PlatformIdentifier
Manifest   1.0.0.0    PnpDevice                           {Get-PnpDevice, Get-PnpDeviceProperty, Enable-PnpDevice, Disab
Manifest   1.1        PrintManagement                     {Add-Printer, Add-PrinterDriver, Add-PrinterPort, Get-PrintCon
Manifest   1.1        PSDesiredStateConfiguration         {Set-DscLocalConfigurationManager, Start-DscConfiguration, Tes
Script     1.0.0.0    PSDiagnostics                       {Disable-PSTrace, Disable-PSWSManCombinedTrace, Disable-WSManT
Binary     1.1.0.0    PSScheduledJob                      {New-JobTrigger, Add-JobTrigger, Remove-JobTrigger, Get-JobTri
Manifest   2.0.0.0    PSWorkflow                          {New-PSWorkflowExecutionOption, New-PSWorkflowSession, nwsn}
Manifest   1.0.0.0    PSWorkflowUtility                   Invoke-AsWorkflow
Manifest   3.0.0.0    RemoteAccess                        {Add-DAAppServer, Add-DAClient, Add-DAClientDnsConfiguration,
Manifest   2.0.0.0    RemoteDesktop                       {Get-RDCertificate, Set-RDCertificate, New-RDCertificate, New-
Manifest   2.0.0.0    RemoteDesktopServices               Convert-License
Manifest   1.0.0.0    ScheduledTasks                      {Get-ScheduledTask, Set-ScheduledTask, Register-ScheduledTask,
Manifest   2.0.0.0    SecureBoot                          {Confirm-SecureBootUEFI, Set-SecureBootUEFI, Get-SecureBootUEF
Manifest   1.0.0.0    SecurityCmdlets                     {Backup-SecurityPolicy, Restore-SecurityPolicy, Backup-AuditPo
Script     1.0.0.0    ServerCore                          {Get-DisplayResolution, Set-DisplayResolution}
Script     2.0.0.0    ServerManager                       {Get-WindowsFeature, Install-WindowsFeature, Uninstall-Windows
Cim        1.0.0.0    ServerManagerTasks                  {Get-SMCounterSample, Get-SMPerformanceCollector, Start-SMPerf
Manifest   2.0.0.0    SmbShare                            {Get-SmbShare, Remove-SmbShare, Set-SmbShare, Block-SmbShareAc
Manifest   2.0.0.0    SmbWitness                          {Get-SmbWitnessClient, Move-SmbWitnessClient, gsmbw, msmbw...}
Manifest   2.0.0.0    SoftwareInventoryLogging            {Get-SilComputer, Get-SilComputerIdentity, Get-SilSoftware, Ge
Manifest   1.0.0.0    StartLayout                         {Export-StartLayout, Import-StartLayout, Get-StartApps}
Manifest   2.0.0.0    Storage                             {Add-InitiatorIdToMaskingSet, Add-PartitionAccessPath, Add-Phy
Manifest   2.0.0.0    TLS                                 {New-TlsSessionTicketKey, Enable-TlsSessionTicketKey, Disable-
Manifest   1.0.0.0    TroubleshootingPack                 {Get-TroubleshootingPack, Invoke-TroubleshootingPack}
Manifest   2.0.0.0    TrustedPlatformModule               {Get-Tpm, Initialize-Tpm, Clear-Tpm, Unblock-Tpm...}
Binary     2.1.639.0  UEV                                 {Clear-UevConfiguration, Clear-UevAppxPackage, Restore-UevBack
Manifest   1.0.0.0    UserAccessLogging                   {Enable-Ual, Disable-Ual, Get-Ual, Get-UalDns...}
Manifest   2.0.0.0    VpnClient                           {Add-VpnConnection, Set-VpnConnection, Remove-VpnConnection, G
Manifest   1.0.0.0    Wdac                                {Get-OdbcDriver, Set-OdbcDriver, Get-OdbcDsn, Add-OdbcDsn...}
Manifest   1.0.0.0    WebAdministration                   {Start-WebCommitDelay, Stop-WebCommitDelay, Get-WebConfigurati
Manifest   2.0.0.0    Whea                                {Get-WheaMemoryPolicy, Set-WheaMemoryPolicy}
Manifest   1.0.0.0    WindowsDeveloperLicense             {Get-WindowsDeveloperLicense, Unregister-WindowsDeveloperLicen
Script     1.0        WindowsErrorReporting               {Enable-WindowsErrorReporting, Disable-WindowsErrorReporting,
Manifest   1.0.0.0    WindowsSearch                       {Get-WindowsSearchSetting, Set-WindowsSearchSetting}
Manifest   1.0.0.0    WindowsUpdate                       Get-WindowsUpdateLog


    Directory: C:\Program Files\Microsoft Monitoring Agent\Agent\PowerShell


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Binary     1.0.0.0    Microsoft.MonitoringAgent.PowerS... {Checkpoint-WebApplicationMonitoring, Get-WebApplicationMonito


PS C:\Users\xxx> Get-InstalledModule

Version    Name                                Repository           Description
-------    ----                                ----------           -----------
2.0.2.128  AzureAD                             PSGallery            Azure Active Directory V2 General Availability Modul
1.0.957478 AzureADExporter                     PSGallery            This module exports all the Azure AD objects and Ide
2.0.2.129  AzureADPreview                      PSGallery            Azure Active Directory V2 Preview Module. ...
20.12.4... BinaryTree.Power365                 PSGallery            PowerShell Management cmdlets for Binary Tree Power3
1.3.0.6    DSCParser                           PSGallery            This module allows for the parsing of a DSC Configur
3.1.0      ExchangeOnlineManagement            PSGallery            This is a General Availability (GA) release of the E
0.11.0     ExchangePowerShell                  PSGallery            The EP (ExchangePowerShell) Powershell module is a c
7.1.1      ImportExcel                         PSGallery            PowerShell module to import/export Excel spreadsheet
1.21.0     Microsoft.Graph                     PSGallery            Microsoft Graph PowerShell module
0.9.1      Microsoft.Graph.Analytics           PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Applications        PSGallery            Microsoft Graph PowerShell Cmdlets
1.23.0     Microsoft.Graph.Authentication      PSGallery            Microsoft Graph PowerShell Authentication Module.
1.21.0     Microsoft.Graph.Bookings            PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Calendar            PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.ChangeNotifications PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.CloudCommunications PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Compliance          PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.CrossDeviceExper... PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.DeviceManagement    PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.DeviceManagement... PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.DeviceManagement... PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.DeviceManagement... PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.DeviceManagement... PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Devices.CloudPrint  PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Devices.Corporat... PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Devices.ServiceA... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.DevicesApps.Devi... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.DevicesApps.Mobi... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.DevicesApps.Offi... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.DevicesApps.Shar... PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.DirectoryObjects    PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Education           PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Files               PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Files.Drives        PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Files.Permissions   PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Files.Shares        PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Financials          PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Groups              PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Actions      PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Calendar     PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Conversation PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Conversat... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Directory... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Drive        PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Endpoint     PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Extension    PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Functions    PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Group        PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Lifecycle... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.OneNote      PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Planner      PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.ProfilePhoto PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Groups.Site         PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.AccessR... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Adminis... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Applica... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.AppRole... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.AuditLogs  PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Authent... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.AzureADPIM PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Conditi... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Contracts  PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.DataPol... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Devices    PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Directory  PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Identity.Directo... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Directo... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Directo... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Directo... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Directo... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Directo... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Domains    PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Identity.Governance PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Invitat... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.OAuth2P... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.OnPremi... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Organiz... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Organiz... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Policies   PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Protection PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Providers  PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.RoleMan... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Service... PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Identity.SignIns    PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.Subscri... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.TermsOfUse PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.TrustFr... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Identity.UserFlows  PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Mail                PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.ManagedTenants      PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Notes               PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Notification        PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.OnlineMeetings      PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.People              PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.PersonalContacts    PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Places              PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Planner             PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Reports             PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.SchemaExtensions    PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Search              PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Security            PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Sites               PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Sites.Actions       PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Sites.Drive         PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Sites.Functions     PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Sites.List          PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Sites.OneNote       PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Sites.Pages         PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Sites.Site          PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Subscriptions       PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Teams               PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Teams.AppCatalogs   PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Teams.Chats         PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Teams.Team          PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Teams.Teamwork      PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Users               PSGallery            Microsoft Graph PowerShell Cmdlets
1.22.0     Microsoft.Graph.Users.Actions       PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.ActivityFeed  PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Calendar      PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Contacts      PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Devices       PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.DirectoryO... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Drive         PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Extensions    PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.FollowedSites PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.Users.Functions     PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Groups        PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Informatio... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.LicenseDet... PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Mail          PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.OneNote       PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.OutlookUser   PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.People        PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.Planner       PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.ProfilePhoto  PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.User          PSGallery            Microsoft Graph PowerShell Cmdlets
0.9.1      Microsoft.Graph.Users.UserSettings  PSGallery            Microsoft Graph PowerShell Cmdlets
1.21.0     Microsoft.Graph.WindowsUpdates      PSGallery            Microsoft Graph PowerShell Cmdlets
16.0.22... Microsoft.Online.SharePoint.Powe... PSGallery            Microsoft SharePoint Online Services Module for Wind
2.0.155    Microsoft.PowerApps.Administrati... PSGallery            PowerShell interface for Microsoft PowerApps and Flo
1.23.308.1 Microsoft365DSC                     PSGallery            This DSC module is used to configure and monitor Mic
5.0.0      MicrosoftTeams                      PSGallery            Microsoft Teams cmdlets module for Windows PowerShel
1.0.105    MSCloudLoginAssistant               PSGallery            Checks the current status of connections to (and as
1.3.0.0    MSGraph                             PSGallery            Tools for interacting with the Microsoft Graph Api
1.1.183.57 MSOnline                            PSGallery            Microsoft Azure Active Directory Module for Windows
1.0.51     MSOnlineExt                         PSGallery            This PowerShell module was made to ease the burden o
1.12.0     PnP.PowerShell                      PSGallery            Microsoft 365 Patterns and Practices PowerShell Cmdl
1.7.249    PSFramework                         PSGallery            General Scripting Framework, providing PowerShell-sp
2.0.0.14   ReverseDSC                          PSGallery            This DSC module is used to extract the DSC Configura
1.4.7      PackageManagement                   PSGallery            PackageManagement (a.k.a. OneGet) is a new way to di

**Comment by andikrueger** [comment]
Somehow there is in an issue with the older version of the various Microsoft.Graph.XZY Modules. These need to be removed. As a final result, only the latest version required by M365DSC should be available.

This list contains the modules needed by M365DSC> https://github.com/microsoft/Microsoft365DSC/blob/master/Modules/Microsoft365DSC/Dependencies/Manifest.psd1

Please do close and reopen your PowerShell session between the commands.
